client/my-sites/hosting/controller.js

```diff
@@ -0,0 +1,33 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import config from 'config';
+import page from 'page';
+
+/**
+ * Internal Dependencies
+ */
+import Hosting from './main';
+import isSiteAutomatedTransfer from 'state/selectors/is-site-automated-transfer';
+import { getSelectedSiteId } from 'state/ui/selectors';
+
+export function redirectIfNotAtomic( context, next ) {
+	const { store } = context;
+	const state = store.getState();
+	const isAtomic = isSiteAutomatedTransfer( state, getSelectedSiteId( state ) );
+
+	if ( ! config.isEnabled( 'hosting' ) || ! isAtomic ) {
```

Comment:
Minor, but we can use another early return above to avoid some store checks.

For example:

```jsx
if ( ! config.isEnabled( 'hosting' ) {
    return page.redirect( '/' ); //can bail out of the routing chain
}
const { store } = context;
const state = store.getState();
const isAtomic = isSiteAutomatedTransfer( state, getSelectedSiteId( state ) );

if ( ! isAtomic ) {
    return page.redirect( '/' );
}

next();
```

Reply:
is redirecting to `/` appropriate? it seems like the only way we could get here is to _either_ manually enter the address or follow a link _or_ be at the hosting panel and switch sites.

in either case it seems like it might be confusing to redirect to another panel. would it be appropriate instead to show a notice that there are no hosting settings available and oh-by-the-way we have this plan available where you can do that?

in such a case I could imagine seeing code here more reminiscent of the editor loading, assuming a new `/hosting` API endpoint which provides the information.

sorry for the long comment; this follows a chat in Slack. please ignore at whim or ask for clarification without hesitation or discuss

```js
export function layout( content, next ) {
	const { store } = context;

	// the siteId may not be available immediately if we directly navigated here
	const hostingInfo = waitForSiteId( store.getState )
		.then( siteId => waitForData( {
			hostingInfo: requestHostingInfo( siteId )
		} ) )

	context.primary = React.createElement( asyncLoader( {
		promises: {
			data: hostingInfo
		},
		loading: () => <div>Loading…</div>,
		success: () => <HostingPanel />,
		failure: () => <div>Could not get hosting information, sorry</div>
	} ) )
}

const HostingPanel = connect( ( state, { siteId } ) => {
	hostingInfo: requestHostingInfo( siteId ),
} )( { hostingInfo } ) => {
	switch ( hostingInfo.status ) {
		case NO_HOSTING_INFO:
			return <NoHostingPanel reason={ hostingInfo.reason } />;

		case NO_PERMISSION:
			return <OopsYouCannotSeeMe />;

		case HOSTING_INFO_VALID:
			return <HostingInfo info={ hostingInfo.deets } />;
	}
}

// this may or may not already exist somewhere already
function waitForSiteId( getState ) {
	return new Promise( resolve => {
		const state = getState();
		const siteId = getSelectedSiteId( state );

		if ( ! siteId ) {
			return;
		}

		resolve( siteId );
	} );
}
```

Reply:
We'll eventually show this section for everyone, but right now phase 1 only focuses on showing this section for Atomic sites. Phase 2 includes folks who can transfer, and phase 3 shows this section for all sites.

Reply:
> We'll eventually show this section for everyone, but right now phase 1 only focuses on showing this section for Atomic sites.

Yes, to confirm, this is one of the reasons that led me to suggest we keep `isAtomic` tests out of Calypso and fold them into a new API endpoint on the backend which can return a declaration that the hosting information is available (and supplied with the sftp/mysql credentials) or unavailable for some reason (such as because it's not relevant to the site or because the user lacks permissions)

Or at least, write this code assuming that API exists so that we can iterate on the visual elements before that mechanism is in place

```js
dispatch( {
  type: 'HOSTING_INFO_SET',
  siteId: 12345,
  status: 'hosting_information_valid',
  sftp: {
    username, password, url, port
  },
  phpmyadmin: 'https://…'
} )

dispatch( {
  type: 'HOSTING_INFO_SET',
  siteId: 1234,
  status: 'hosting_information_unavailable',
  reason: 'missing_permission'
} )
```

Etc…

Reply:
Yup, I think that's the way to go, but I will add that in in my next PR. 👍 Thanks, Dennis!